core/merginapi.cpp

```diff
@@ -210,6 +212,19 @@ QString MerginApi::listProjectsByName( const QStringList &projectNames )
     return QLatin1String();
   }
 
+  const int listProjectsByNameApiLimit = 50;
+  QStringList projectNamesToRequest( projectNames );
+
+  if ( projectNamesToRequest.count() > listProjectsByNameApiLimit )
+  {
+    CoreUtils::log( "list projects by name", QStringLiteral( "Too many local projects: " ) + QString::number( projectNames.count(), 'f', 0 ) );
+    const int projectsToRemoveCount = projectNames.count() - listProjectsByNameApiLimit;
+    QString msg = tr( "Reached limit of local projects.\nRemove %1 projects from your device!" ).arg( projectsToRemoveCount );
```

Comment:
let's rephrase...

e.g. Please remove some projects as the app currently only allows up to 50 downloaded projects.